### PR TITLE
GDS Admin API: Adds audience and subject to tokens

### PR DESCRIPTION
**Pull request by bbengfort**
This PR adds aud and sub claims to access and refresh tokens and verifies the aud is correct. This will prevent a token issued by a trisatest.net server to be used by a vaspdirectory.net server and improves the claims and verification model of admin API authentication. The audience needs to be configured on the server, which required a change to some of the previous configuration values; these configuration changes also affected docker-compose. When we redeploy this API we'll need to make sure that the new configurations are updated. 

Fixes SC-1059

### CODE REVIEW

File: containers/.env.template [hunk 1]
```diff
@@ -15,7 +15,7 @@ SENDGRID_API_KEY=SG.notarealapikey
 
 # Google Identity Services
 # These two environment variables should be identical
-GDS_ADMIN_AUDIENCE=""
+GDS_ADMIN_OAUTH_GOOGLE_AUDIENCE=""
```
Comment (bbengfort) [inline containers/.env.template:18]: Renamed for clarity

File: pkg/gds/config/config.go [hunk 2]
```diff
@@ -53,6 +53,11 @@ type AdminConfig struct {
 	TokenKeys map[string]string `split_words:"true"`
 }
 
+type OauthConfig struct {
+	GoogleAudience         string   `split_words:"true"`
+	AuthorizedEmailDomains []string `split_words:"true"`
+}
+
```
Comment (bbengfort) [inline pkg/gds/config/config.go:60]: Moved Oauth configuration away from admin configuration to make it more understandable.

File: pkg/gds/config/config.go [hunk 3]
```diff
@@ -38,13 +38,13 @@ type GDSConfig struct {
...
-	Audience          string   `split_words:"true"`
-	AuthorizedDomains []string `split_words:"true"`
-	AllowOrigins      []string `split_words:"true" default:"http://localhost,http://localhost:3000,http://localhost:3001"`
-	CookieDomain      string   `split_words:"true"`
+	Enabled      bool     `split_words:"true" default:"true"`
+	BindAddr     string   `split_words:"true" default:":4434"`
+	Mode         string   `split_words:"true" default:"release"`
+	AllowOrigins []string `split_words:"true" default:"http://localhost,http://localhost:3000,http://localhost:3001"`
+	CookieDomain string   `split_words:"true"`
+	Audience     string   `split_words:"true"`
```
Comment (bbengfort) [inline pkg/gds/config/config.go:46]: This `Audience` configuration refers to the token audience claim verification not the Google credential audience, which has been moved to the OauthConfig.

File: pkg/gds/tokens/tokens.go [hunk 4]
```diff
@@ -166,6 +171,7 @@ func (tm *TokenManager) CreateAccessToken(creds interface{}) (_ *jwt.Token, err
 		claims.Email = t.Email
 		claims.Name = t.Name
 		claims.Picture = t.Picture
+		claims.Subject = t.Subject
```
Comment (rebeccabilbro) [inline pkg/gds/tokens/tokens.go:174]: In lines 160-174, might we annotate with comments to remind ourselves which parts of the switch get engaged where?

**Pull request merged**